processing/src/main/java/io/druid/query/aggregation/NullableBufferAggregator.java

```diff
@@ -0,0 +1,91 @@
+/*
+ * Licensed to Metamarkets Group Inc. (Metamarkets) under one
+ * or more contributor license agreements. See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership. Metamarkets licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License. You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package io.druid.query.aggregation;
+
+import io.druid.segment.ColumnValueSelector;
+
+import java.nio.ByteBuffer;
+
+public class NullableBufferAggregator implements BufferAggregator
```

Comment:
I don't know for other types of queries, but at least for topN a more space efficient layout is to keep null flags in a separate bitset (it could be byteBuffer-backed as well)